src/app/SIPDNS/SIPDNSManager.cs

```diff
@@ -240,39 +342,41 @@ public static async Task<SIPDNSLookupResult> ResolveAsync(SIPURI sipURI)
                             else
                             {
                                 // Preference for IPv4 IP address for local host name lookup.
-                                IPAddress firstAddress = addressList.Where(x => x.AddressFamily == AddressFamily.InterNetwork).FirstOrDefault() ?? addressList.FirstOrDefault();
+                                IPAddress firstAddress = addressList.Where(x => x.AddressFamily == (preferIPv6 == true ? AddressFamily.InterNetworkV6 : AddressFamily.InterNetwork)).FirstOrDefault() ?? addressList.FirstOrDefault();
                                 SIPEndPoint resultEp = new SIPEndPoint(sipURI.Protocol, new IPEndPoint(firstAddress, port));
                                 return new SIPDNSLookupResult(sipURI, resultEp);
                             }
                         }
                         else
                         {
-                            return await Task.Run(() => DNSARecordLookup(hostOnly, port, false, sipURI));
+                            return await Task.Run(() => DNSNameRecordLookup(hostOnly, port, false, sipURI, null, preferIPv6));

```

Comment:
Why the await here? It looks like the code can be written so that we can return the Task.

Reply:
because I tend to not change programming style of the original author, when I add functionality